static/src/javascripts/projects/common/modules/crosswords/anagram-helper/main.js

```diff
@@ -0,0 +1,167 @@
+// @flow
+import React from 'react/addons';
+import { inlineSvg } from 'common/views/svgs';
+import ClueInput from 'common/modules/crosswords/anagram-helper/clue-input';
+import CluePreview from 'common/modules/crosswords/anagram-helper/clue-preview';
+import Ring from 'common/modules/crosswords/anagram-helper/ring';
+import helpers from 'common/modules/crosswords/helpers';
+import shuffle from 'lodash/collections/shuffle';
+import rest from 'lodash/arrays/rest';
+
+const AnagramHelper = React.createClass({
+    getInitialState() {
+        return {
+            clueInput: '',
+            showInput: true,
+        };
+    },
+
+    componentWillReceiveProps(next: any) {
+        // reset on clue change
+        if (next.clue !== this.props.focussedEntry) {
+            this.reset();
+        }
+    },
+
+    onClueInput(text: string) {
+        if (!/\s|\d/g.test(text)) {
+            this.setState({
+                clueInput: text,
+            });
+        }
+    },
+
+    /**
+     * Shuffle the letters in the user's input.
+     *
+     * First, create an array of input characters that have already been entered
+     * into the grid. Then build a new collection of letters, using the first
+     * array to flag letters that are already entered in the puzzle, and
+     * shuffle it.
+     *
+     * @param  {String}   word     word to shuffle
+     * @param  {[Object]} entries  array of entries (i.e. grid cells)
+     * @return {[Object]}          array of shuffled letters
+     */
+    shuffleWord(word: string, entries: Array<any>) {
```

Comment:
I checked the spec that tests out the shuffleWord function and it's doing this:

                var entries = testCase.entries.map(function (e) {
                    return { value: e };
                });
                var result = helper.shuffleWord(testCase.word, entries);

So I guess I might be able to have the param type be Array<{value: string}>.

Will try it out